Switch alignment and style adjustments

landitus opened the pull request:
- [x] ⭐️💅Fix vertical alignment of witch with label in Form.Field   
- [x] 💅Fix text style for Label to use the proper uppercase style

siddharthkp commented on src/components/atoms/switch/index.js:
```diff
@@ -4,6 +4,11 @@ import PropTypes from 'prop-types'
+const StyledSwitch = styled.span`
+  display: inline-block;
+  vertical-align: middle;
+  height: 32px;
```
Should this height come from a common place?

siddharthkp commented on src/components/atoms/switch/index.js:
```diff
@@ -52,6 +57,9 @@ const Toggle = styled.span`
 const Label = styled.label`
   vertical-align: top;
   line-height: 32px;
+  font-size: 13px;
```
tokenssss

siddharthkp commented on src/components/molecules/form/field/index.js:
```diff
@@ -28,6 +29,9 @@ const StyledField = styled.div`
     /* TO-DO: Tokenize this value as it's related to the other heights of components. */
   }
+  ${StyledSwitch} {
+    margin-top: 6px;
```
This is a non standard value 😢

landitus replied:
I had to make this adjustment when the switch is in the Field (left-aligned) it needs this notch of margin.

landitus replied:
I think we need to review this in the top-align version. I'll add a big form in the overview to test all fields with top alignment.

landitus commented:
It could. I had to declare the height to get it to align. In this case the height is part of the scale, but I don’t think we should use a spacing token here. 

> On 8 Feb 2018, at 09:39, Siddharth Kshetrapal <notifications@github.com> wrote:
> 
> @siddharthkp commented on this pull request.
> 
> In src/components/atoms/switch/index.js:
> 
> > @@ -4,6 +4,11 @@ import PropTypes from 'prop-types'
>  
>  import { colors, spacing, misc } from '../../../tokens/'
>  
> +const StyledSwitch = styled.span`
> +  display: inline-block;
> +  vertical-align: middle;
> +  height: 32px;
> Should this height come from a common place?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

landitus commented:
> We can add a todo to tokenise them later

I agree.

> There's a 6px margin which is a non standard value 😢, can we make it 8?

that 6px corresponds (44-32)/2 so I could express it an operation between variables.

siddharthkp commented:
> that 6px corresponds (44-32)/2 so I could express it an operation between variables

umm, that's seems forced. Can we get away with away with `margin-top: 8px;`? If it's internal to the component, we can have non-standard values. I think we have some of those to solve alignment issues with select/textarea

landitus commented:
Yes, this is that case where it's just internal for this component. Any other value makes the switch look off center.

siddharthkp commented:
I think you just need to tokenise font size and we're good!

landitus commented:
awesome! will do 

> On Feb 9, 2018, at 01:04, Siddharth Kshetrapal <notifications@github.com> wrote:
> 
> I think you just need to tokenise font size and we're good!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/auth0/cosmos/pull/226#issuecomment-364329236>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAOmbxB98TAMEnNtKM7egPHJXV5k_AnOks5tS8PcgaJpZM4R9m_t>.
>

landitus commented:
I've added the tokens, and also fixed a small spacing that the hidden `<input>` before the Switch was creating. The solution is to remove the element from the DOM flow. I've avoided using `display: none` because maybe it affects the form or accessibility. Did a position absolute instead. But I'm open to ideas of the best way to solve this.
